devito/propagator.py

```diff
@@ -47,11 +50,12 @@ class Propagator(object):
                            Set cache_blocking to AutoTuner instance, to use auto tuned
                            tune block sizes
     """
-    def __init__(self, name, nt, shape, stencils, factorized=None, spc_border=0,
-                 time_order=0, time_dim=None, space_dims=None, dtype=np.float32,
-                 forward=True, compiler=None, profile=False,
+    def __init__(self, name, nt, shape, stencils=None, loop_body=None, factorized=None,
```

Comment:
In principle I am not opposed to a C statement backdoor (3rd-tier API), but I don't think that bolting this onto the Propagator with yet another constructor argument that conceptually conflicts with the main `stencils` argument is the way to do this. I am also not yet convinced that the current problem we're discussing in this PR warrants this extension, since we're just trying to get around current limitations of the low-level (2nd-tier) symbolic API in the current Propagator - and I'd prefer we fix the problem, rather than add more complexity to work around it.